000-mm2-connector-offsets-support.md

```diff
@@ -0,0 +1,449 @@
+
+## Affected/not affected projects
+
+This proposal only affects the KafkaMirrorMaker2 parts of the cluster-operator.
+It has been designed to work in a similar way to [proposal #76][proposal-76] so that code can be shared with KafkaConnector.
+
+## Compatibility
+
+N/A
+
+## Rejected alternatives
+
+### Applying actions to multiple connectors
+
+Instead of making the `strimzi.io/mirrormaker-connector` required, we could make it optional or a comma-separated list.
+
+In the case where it was optional, if only the `strimzi.io/connector-offsets` annotation is present, the action would be applied to all mirroring routes and all connectors within mirroring routes that are configured in the KafkaMirrorMaker2 resource.
+If the `strimzi.io/mirrormaker-connector` annotation is also present, the action would only be applied to the connector identified in the annotation.
+
+In the case where it was a comma-separated list it would be required and explicitly list the connectors to apply the action to.
+
+This would be a nice feature to allow users to quickly list, alter and reset offsets for many connectors at once.
+Particularly for listing and resetting offsets this seems very useful.
+
+However it will add a lot of complexity because the operator reconciles one connector at a time.
+This means we would need to introduce a mechanism to keep track of when we can remove the annotations and deal with error cases if for example one connector is stopped, but another isn't.
+
+### Validating the structure of alterOffsets ConfigMap
+
+A previous version of this proposal described more complex validation of the JSON provided for altering offsets.
+This included validating the specific fields that are needed, and even not requiring fields that the operator could infer.
+
+However, this would make the operator vulnerable to changes in the way the MirrorMaker connectors store their offsets so was rejected in favour of less strict validation.
+
+### Preventing the user from altering or reseting offsets for the MirrorCheckpointConnector and MirrorHeartbeatConnector
+
+The MirrorCheckpointConnector and MirrorHeartbeatConnector both store offsets in Kafka, however they do not actually use those offsets.
+MirrorMaker users may still track offsets to track the progress of those connectors, but since they are not used, it isn't immediately obvious why a user would want to reset or alter them.
+This [Apache Kafka PR](https://github.com/apache/kafka/pull/14005) talks about some of the reasons why these operations may still be desired by users.
+In addition, it is possible that in future these connectors will make use of these offsets.
+For these reasons this proposal explicitly allows users to perform list/alter/reset on all MirrorMaker connectors.
```

Comment:
I wonder if it's still worth documenting this behavior as I expect some users may try to use these operations.

Reply:
Yeah we can certainly document that we would only recommend running these operations for MirrorSourceConnector